src/pipecat/utils/string.py

```diff
@@ -17,9 +18,36 @@
     (\。\s*\。\s*\。|[。？！；।])  # the full-width version (mainly used in East Asian languages such as Chinese, Hindi)
     $                # End of string
 """
+
 ENDOFSENTENCE_PATTERN = re.compile(ENDOFSENTENCE_PATTERN_STR, re.VERBOSE)
 
+EMAIL_PATTERN = re.compile(r"[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}")
+
+NUMBER_PATTERN = re.compile(r"[+-]?(\d+(\.\d*)?|\.\d+)([eE][+-]?\d+)?")
+
+
+def replace_match(text: str, match: re.Match, old: str, new: str) -> str:
+    start = match.start()
+    end = match.end()
+    replacement = text[start:end].replace(old, new)
+    text = text[:start] + replacement + text[end:]
+    return text
+
 
 def match_endofsentence(text: str) -> int:
-    match = ENDOFSENTENCE_PATTERN.search(text.rstrip())
+    text = text.rstrip()
+
+    # Replace email dots by ampersands so we can find the end of sentence.
```

Comment:
Really nit: Would be worth to add a small example in the comment, just to make easier to understand.
For example:
- this email: first.last@email.com
- will be replaced by this: first&last@email&com

Reply:
Done!